### PR TITLE
Add negative matchers for inhibitors

### PR DESCRIPTION
**Pull request by aSquare14**
Related issue : #1023
Related Pull Requests : #2456 #2434 #2441 
Signed-off-by: aSquare14 <atibhi.a@gmail.com>

### CODE REVIEW

**Comment by aSquare14** [comment]
cc @beorn7 @vladimiroff @simonpasquier  Could you please review this.

**Review by beorn7 — CHANGES_REQUESTED** [review]
I think this works. Just a few nits about the doc comments.

File: config/config.go [hunk 1]
```diff
@@ -645,17 +645,23 @@ func (r *Route) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
 	// SourceMatch defines a set of labels that have to equal the given
-	// value for source alerts.
+	// value for source alerts. Deprecated. Remove before v1.0 release.
 	SourceMatch map[string]string `yaml:"source_match,omitempty" json:"source_match,omitempty"`
 	// SourceMatchRE defines pairs like SourceMatch but does regular expression
-	// matching.
+	// matching. Deprecated. Remove before v1.0 release.
 	SourceMatchRE MatchRegexps `yaml:"source_match_re,omitempty" json:"source_match_re,omitempty"`
+	// SourceMatchers defines a set of labels that have to be equal or not equal to the given
+	// value for source alerts. It also does regular expression matching.
```
[the remark this answers is not in the document]
Comment (aSquare14) [reply config/config.go:654]: Done

**Review by beorn7 — APPROVED** [review]
Great, thank you.

**Comment by beorn7** [comment]
I got in touch with @simonpasquier , BTW. He is really spread thin at the moment. It will get better, but we also try to find a co-maintainer for him. In the meantime, I'll merge PRs where I'm reasonably confident. Like this one. (o:

**Comment by aSquare14** [comment]
Thanks for merging and the reviews :)

**Pull request merged**